sdk/resourcemanager/Azure.ResourceManager/tests/Samples/AuxiliaryPoilcy.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.ResourceManager.Tests.Samples
+{
+    #region Snippet:Sample_Header_Policy
+    internal class AuxiliaryPoilcy : HttpPipelineSynchronousPolicy
+    {
+        private static string AUTHORIZATION_AUXILIARY_HEADER = "x-ms-authorization-auxiliary";
+        string _token;
+
+        internal AuxiliaryPoilcy(string token)
+        {
+            _token = token;
+        }
+
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            string token = "Bearer " + _token;
+            if (!message.Request.Headers.TryGetValue(AUTHORIZATION_AUXILIARY_HEADER, out _))
```

Comment:
we could use `ContainsKey` instead if we do not want the value

Reply:
The `Request.Headers` donesn't have `ContainsKey()` method, we could stay on this for now.